rxandroid/src/main/java/rx/android/lifecycle/LifecycleHelper.java

```diff
@@ -0,0 +1,92 @@
+package rx.android.lifecycle;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import rx.subjects.BehaviorSubject;
+
+/**
+ * This instance lives within the scope of an {@code Activity} in order to forward Lifecycle events
+ * to {@link rx.subjects.BehaviorSubject}
+ *
+ * Thread safe, since it's <i>Thread-confined</i> to the main Thread.
+ *
+ * @author Nabil Hachicha.
+ */
+class LifecycleHelper implements Application.ActivityLifecycleCallbacks {
+    private final Activity activityToMonitor;
+    private final BehaviorSubject<LifecycleEvent> lifecycleSubject;
```

Comment:
Why is this a BehaviorSubject. I'm relatively new to RX but won't this begin by emitting the last lifecycle event received and then any other lifecycle events it receives. Is that the behaviour you want? Is this to catch the case when someone subscribes after onDestroy?